src/components/comments/html-pin-adapter/index.ts

```diff
@@ -0,0 +1,884 @@
+import { Logger, Observer } from '../../../common/utils';
+import { PinMode } from '../../../web-components/comments/components/types';
+import { Annotation, PinAdapter, PinCoordinates } from '../types';
+
+import {
+  HorizontalSide,
+  TranslateAndScale,
+  Simple2DPoint,
+  SimpleParticipant,
+  TemporaryPinData,
+} from './types';
+
+export class HTMLPin implements PinAdapter {
+  // Public properties
+  // Observers
+  public onPinFixedObserver: Observer;
+
+  // Private properties
+  // Comments data
+  private annotations: Annotation[];
+  private localParticipant: SimpleParticipant = {};
+
+  // Loggers
+  private logger: Logger;
+
+  // Booleans
+  private isActive: boolean;
+  private isPinsVisible: boolean = true;
+  private movedTemporaryPin: boolean;
+
+  // Data about the current state of the application
+  private selectedPin: HTMLElement | null = null;
+  private dataAttribute: string = 'data-superviz-id';
+
+  // Coordinates/Positions
+  private mouseDownCoordinates: Simple2DPoint;
+  private commentsSide: HorizontalSide = 'left';
+  private temporaryPinCoordinates: TemporaryPinData = {};
+
+  // Elements
+  private container: HTMLElement;
+  private pinsContainer: HTMLDivElement;
+  private elementsWithDataId: Record<string, HTMLElement> = {};
+  private divWrappers: Map<string, HTMLElement> = new Map();
+  private pins: Map<string, HTMLElement>;
+  private traversedDom: Set<HTMLElement> = new Set();
+
+  // Observers
+  private resizeObserver: ResizeObserver;
+  private mutationObserver: MutationObserver;
+
+  constructor(containerId: string, dataAttributeName?: string) {
```

Comment:
```suggestion
  constructor(containerId: string,  { dataAttributeName }: HTMLPinOptions) {
```

We may need more properties here, to prevent break changes, must be an object.